src/lib/es-mapping-ts/es-mapping.ts

```diff
@@ -37,7 +37,17 @@ export class InternalEsMapping {
 
   addProperty(name: string | symbol, mapping: InternalEsMappingProperty): void {
     this.properties.set(name, mapping);
-    this.esmapping.body.properties[name] = mapping.propertyMapping;
+
+    if (!mapping.propertyMapping) {
+      return;
+    }
+
+    const propertyMapping = JSON.parse(JSON.stringify(mapping.propertyMapping));
```

Comment:
I suppose you use this to make a clone.  I chose to use lodash in the library, so can you just change this by  `lodash.deepClone(mapping.propertyMapping)` it will be more understandable

Reply:
🔧 fixed

In general I avoid using `cloneDeep` based on https://github.com/lodash/lodash/issues/3676. `JSON.parse(JSON.stringify(...))` is slower but meh most of the time :D